### ISSUE TITLE
A different generic_hygrostat getting merged into HA core

### ISSUE THREAD

**Issue by ltjessem** [Corsw/homeassistant-generic-hygrostat#19]
Looks like generic_hygrostat (not this one, a less functional one) is getting merged into HA core soon. 
https://github.com/home-assistant/core/pull/36759

Do we need to change the name of the integration in preparation for this?

**Comment by Knodd** [comment]
It seems the other generic hygrostat is more of a humidifier controller with fixed setpoints. I use this integration to compensate for humidity surges when showering by increasing ventilation.

If the other one is merged I don’t see why this shouldn’t be also and a name change might make it easier to convince the bosses.

I believe the reason for merge rejection last time was “you can do this with an automation”, but with the focus now being ease of use and GUI setup I think we should have another go at it.
I researched how to set up this integrations functionality with automations and I never got it working

**Comment by Knodd** [comment]
How about changing name to “humidity controlled ventilation” or something similar?

**Comment by KTibow** [comment]
IMO since you need to manually create an automation, a better name is "humidity_swing_sensor".

**Comment by Knodd** [comment]
I agree.
This integration should (from a user experience point of view) have an optional setting for a switch entity to turn on /off included and if you want it to do something else an automation could be used.
Might have to try to do a PR when I have some time

**Comment by bartve66** [comment]
Maybe I am stirring up an old discussion, but today I finally switched from a manual installed custom component of this hygrostat to a HACS managed version (so it will be easier to install updates etc.).
I put some effort into evaluating the various options, including the hygrostat that is part of the HA core. I came to the conclusion this one (the HACS variant) is better because of the dynamic/adapting  limits in comparison to the static limits of the HA core variant. Am I right? And if so, why is the better option not integrated into the HA core?

**Comment by ltjessem** [comment]
You are correct. The HA variant is pretty useless as humidity is such a variable thing throughout the year, varying weather conditions etc., you would need to be adjusting the setpoint relatively often. 
As for getting integrated into HA core, from what I've heard that's a difficult needle to thread. In some cases the HA devs will allow the most ridiculous and useless (breaking) changes, and in other cases actual useful stuff like this is denied for arbitrary reasons.
"You can do this with an automation", then they add the most simplistic implementation themselves, that can be done with the most basic of automations. Go figure.

**Comment by ndbroadbent** [comment]
I also agree that this integration is much better, but it probably does need to be renamed now, since the name conflicts with the one in HA.

**Comment by elupus** [comment]
Yup, this need to be renamed. We already have had conflicts reported with confused users.

**Comment by Warp95** [comment]
This integration is much better, but it probably does need to be renamed , since the name conflicts with the one in HA, and I would like to continue using this integration.

**Comment by thomasvnl** [comment]
@elupus yes, I just was one of these confused users. Renaming would be handy, but if that is not really handy in the short term at least add something to the homepage of this repository (readme) as I had to find out by reading this specific issue why the in-built generic hygrostat was not the latest version of this repository (and thus was not what I was expecting).

**Comment by Corsw** [comment]
Thanks for raising this — a very late update, but this is still relevant.

The domain collision with Home Assistant Core's Generic Hygrostat is now a known and documented limitation.

For v0.9.0 I deliberately kept the historical `generic_hygrostat` domain to avoid breaking existing YAML installations. The two integrations serve different purposes:

- This custom integration detects a rapid rise in humidity and exposes a binary sensor.
- Home Assistant Core's Generic Hygrostat maintains a target humidity and controls a humidifier/dehumidifier.

Changing the domain would require a proper migration path for existing users, so I don't want to do that as part of a compatibility-focused release.

This may be revisited in a future release, but for now the existing domain will remain in place.

Thanks again for flagging this originally.
